### ISSUE TITLE
Rust codegen results in clippy warnings

### ISSUE THREAD

**Issue by connorsmith256** [wasmCloud/weld#136]
As of rust 1.67 (possibly slightly earlier), code generated by weld doesn't satisfy default clippy warnings. The most common violation is https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args

Example, from the `messaging` interface:
```
❯ cargo clippy
   Compiling wasmcloud-interface-messaging v0.8.1 (/Users/connor/Documents/wasmCloud/interfaces/messaging/rust)
warning: variables can be used directly in the `format!` string
   --> messaging/rust/src/messaging.rs:484:50
    |
484 |                     .map_err(|e| RpcError::Deser(format!("'SubMessage': {}", e)))?;
    |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args
    = note: `#[warn(clippy::uninlined_format_args)]` on by default
help: change this to
    |
484 -                     .map_err(|e| RpcError::Deser(format!("'SubMessage': {}", e)))?;
484 +                     .map_err(|e| RpcError::Deser(format!("'SubMessage': {e}")))?;
    |
```

This generates noise for everyone, and for users whose CI builds fail on warnings, this will cause failures

**Comment by brooksmtownsend** [comment]
For this one, any of the code in the code generation module that uses format strings should use inlined format arguments. Clippy should be helpful with this one!

**Comment by connorsmith256** [comment]
Fixed by #139
